CollisionDetectionSystem/FunctionalObjects/MockTransponder.cs

```diff
@@ -17,53 +17,86 @@ public class MockTransponder: IMockTransponder
 		// and announce location every 500ms
 		public void Start (String testDirName)  
 		{
-			Console.WriteLine ("testdir="+testDirName);
-			var files = new List<StreamReader>();
+			var dataList = buildTransporterDataList (testDirName);
 
-			string[] txtFiles = Directory.GetFiles(testDirName, "*.txt").Select(path => Path.GetFileName(path)).ToArray(); //array of text files                           
+			while(dataList.Any())
+			{
+				dataList = sendData (dataList);
+			}
+		}
 
-			for (int i = 0; i < txtFiles.Length; i++) //fills list with all streamreaders
+		public List<TransponderData> sendData (List<TransponderData>  dataList )  
+		{
+			var listToSend = new List<TransponderData> ();
+			listToSend.Add (dataList [0]);
+
+			for (int i = 1; i<dataList.Count; i++) 
 			{
-				StreamReader file = new StreamReader (testDirName + "\\" + txtFiles.GetValue (i));
-				files.Add (file);
+				if((dataList [0].Timestamp).Equals (dataList [i].Timestamp))
+				{
+					listToSend.Add (dataList[i]);
+					dataList.RemoveAt(i);
+				}
+				else
+				{
+					break;
+				}
 			}
 
+			dataList.RemoveAt(0);
+
+			BroadcastDataEvent (listToSend);
+
+			return dataList;
+		}
+
+		public List<TransponderData> buildTransporterDataList (String testDirName)  
+		{
+			var files = streamReaders (testDirName);
 
-			String line = "something"; //reading a line of each file at a time... 
+			var dataList = new List<TransponderData> ();
+			String line = "";
 			int count = 1;
+
 			while (files.Any())
 			{
-				for (int i = 0; i < files.Count; i++) 
-				{
+				for (int i = 0; i < files.Count; i++) {
 					if ((line = (files [i]).ReadLine ()) == null) {
 						files.Remove (files [i]);
-					}
-					else if (line.StartsWith ("#") || line.Equals(""))
-					{
+					} else if (line.StartsWith ("#") || line.Equals ("")) {
 						i--;
-					}
-					else 
-					{
+					} else {
 						count++;
-						Console.WriteLine (line + "    " +count/2);
+						//Console.WriteLine (line + "    " + count / 2); // prints line + the line number, ignores lines with #'s
 
 						string[] splitData = line.Split (',');
 
-						if (!(splitData.Length != 6))
-						{
-							
-							TransponderData tData = new TransponderData (splitData [0], splitData [1], double.Parse(splitData [2]), double.Parse(splitData [3]), double.Parse(splitData [4]), splitData [5]);
-							BroadcastDataEvent (tData);
-							System.Threading.Thread.Sleep(500);
```

Comment:
Eh only took about an hour and a half but it's what Steven wanted, basically
it's a lost of all events at a certain time stamp. So all events at a certain
time are sent in together instead of line by line.

I told him we'd agreed to only update when info on our aircraft came in but
yeah....



On Fri, Nov 20, 2015 at 10:13 AM, dsteeve < notifications@github.com [notifications@github.com] > wrote:
In CollisionDetectionSystem/FunctionalObjects/MockTransponder.cs
[https://github.com/dsteeve/teamDrei-4250/pull/19#discussion_r45492854] :

>  
>  						string[] splitData = line.Split (',');
>  
> -						if (!(splitData.Length != 6))
> -						{
> -							
> -							TransponderData tData = new TransponderData (splitData [0], splitData [1], double.Parse(splitData [2]), double.Parse(splitData [3]), double.Parse(splitData [4]), splitData [5]);
> -							BroadcastDataEvent (tData);
> -							System.Threading.Thread.Sleep(500);


calling BroadcastDataEvent(tData) seems like what we wanted to be doing, I still
don't get why you wanted to make this a list of transponderData elements? That
is not how the real transponder would work. The mockTransponder is to mock what
a real transponder would do.

I realize you spent a lot of time to do this, but can you explain why you think
this is the way it should work? What was wrong with the way it was working?

—
Reply to this email directly or view it on GitHub
[https://github.com/dsteeve/teamDrei-4250/pull/19/files#r45492854] .[https://github.com/notifications/beacon/AIfWSsf6TqBYUpvKtmKHK5uLl-kl2Lvlks5pH0xNgaJpZM4GmM_5.gif]